[18.0][FIX] account_banking_pain_base: Fix tests if base_vat + base_bank_from_iban is installed

victoralmau opened the pull request:
Fix tests if `base_vat` is installed

```
File "/opt/odoo/auto/addons/base_vat/models/res_partner.py", line 200, in check_vat
    raise ValidationError(msg)
odoo.exceptions.ValidationError:
The VAT number [BE0123456789] for partner [Test Company] does not seem to be valid. Note: the expected format is BE0477472701
```

Fix tests if `base_bank_from_iban` is installed

```
Traceback (most recent call last):
  File "/opt/odoo/auto/addons/account_banking_pain_base/tests/test_account_backing.py", line 109, in test_except_messages_prepare_field
    partner_bank = self.env["res.partner.bank"].create(
  File "<decorator-gen-208>", line 2, in create
  File "/opt/odoo/custom/src/odoo/odoo/api.py", line 495, in _model_create_multi
    return create(self, [arg])
  File "/opt/odoo/auto/addons/base_bank_from_iban/models/res_partner_bank.py", line 21, in create
    vals_list = [self._add_bank_vals(vals) for vals in vals_list]
  File "/opt/odoo/auto/addons/base_bank_from_iban/models/res_partner_bank.py", line 21, in <listcomp>
    vals_list = [self._add_bank_vals(vals) for vals in vals_list]
  File "/opt/odoo/auto/addons/base_bank_from_iban/models/res_partner_bank.py", line 30, in _add_bank_vals
    vals["bank_id"] = self._get_bank_from_iban(vals["acc_number"]).id
  File "/opt/odoo/auto/addons/base_bank_from_iban/models/res_partner_bank.py", line 36, in _get_bank_from_iban
    iban = schwifty.IBAN(acc_number)
  File "/usr/local/lib/python3.10/site-packages/schwifty/iban.py", line 77, in __init__
    self.validate(validate_bban)
  File "/usr/local/lib/python3.10/site-packages/schwifty/iban.py", line 176, in validate
    self._validate_length()
  File "/usr/local/lib/python3.10/site-packages/schwifty/iban.py", line 189, in _validate_length
    raise exceptions.InvalidLength("Invalid IBAN length")
schwifty.exceptions.InvalidLength: Invalid IBAN length
```

@Tecnativa

victoralmau committed:
[FIX] account_banking_pain_base: Fix tests if base_vat is installed

File "/opt/odoo/auto/addons/base_vat/models/res_partner.py", line 200, in check_vat
    raise ValidationError(msg)
odoo.exceptions.ValidationError:
The VAT number [BE0123456789] for partner [Test Company] does not seem to be valid.
Note: the expected format is BE0477472701

pedrobaeza reviewed (approved):
Other option is to pass the context for not doing the VAT validation.

victoralmau committed:
[FIX] account_banking_pain_base: Fix tests if base_bank_from_iban is installed

Traceback (most recent call last):
  File "/opt/odoo/auto/addons/account_banking_pain_base/tests/test_account_backing.py", line 109, in test_except_messages_prepare_field
    partner_bank = self.env["res.partner.bank"].create(
  File "<decorator-gen-208>", line 2, in create
  File "/opt/odoo/custom/src/odoo/odoo/api.py", line 495, in _model_create_multi
    return create(self, [arg])
  File "/opt/odoo/auto/addons/base_bank_from_iban/models/res_partner_bank.py", line 21, in create
    vals_list = [self._add_bank_vals(vals) for vals in vals_list]
  File "/opt/odoo/auto/addons/base_bank_from_iban/models/res_partner_bank.py", line 21, in <listcomp>
    vals_list = [self._add_bank_vals(vals) for vals in vals_list]
  File "/opt/odoo/auto/addons/base_bank_from_iban/models/res_partner_bank.py", line 30, in _add_bank_vals
    vals["bank_id"] = self._get_bank_from_iban(vals["acc_number"]).id
  File "/opt/odoo/auto/addons/base_bank_from_iban/models/res_partner_bank.py", line 36, in _get_bank_from_iban
    iban = schwifty.IBAN(acc_number)
  File "/usr/local/lib/python3.10/site-packages/schwifty/iban.py", line 77, in __init__
    self.validate(validate_bban)
  File "/usr/local/lib/python3.10/site-packages/schwifty/iban.py", line 176, in validate
    self._validate_length()
  File "/usr/local/lib/python3.10/site-packages/schwifty/iban.py", line 189, in _validate_length
    raise exceptions.InvalidLength("Invalid IBAN length")
schwifty.exceptions.InvalidLength: Invalid IBAN length

pedrobaeza commented:
/ocabot merge patch
